app/code/Magento/Store/Model/CreateStore.php

```diff
@@ -0,0 +1,60 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento\Store\Model;
+
+use Magento\Store\Api\Data\StoreInterface;
+
+/**
+ * Class CreateStore
+ * Model for create a new store view
+ */
+class CreateStore
+{
+    /**
+     * @var \Magento\Store\Api\Data\GroupInterfaceFactory
+     */
+    private $groupFactory;
+
+    /**
+     * @var \Magento\Store\Api\Data\StoreInterfaceFactory
+     */
+    private $storeFactory;
+
+    /**
+     *
+     * @param \Magento\Store\Api\Data\GroupInterfaceFactory $groupFactory
+     * @param \Magento\Store\Api\Data\StoreInterfaceFactory $storeFactory
+     */
+    public function __construct(
+        \Magento\Store\Api\Data\GroupInterfaceFactory $groupFactory,
+        \Magento\Store\Api\Data\StoreInterfaceFactory $storeFactory
+    ) {
+        $this->groupFactory = $groupFactory;
+        $this->storeFactory = $storeFactory;
+    }
+
+    /**
+     *
+     * @param array $data
+     * @return \Magento\Store\Api\Data\StoreInterface
+     */
+    public function execute(array $data): StoreInterface
```

Comment:
It would be better to accept an instance of `StoreInterface` instead of array type that is too general

Reply:
it doesn't have a sense :) because the model loses its meaning 
what now will be doing this model with the name of CreateStore?

little of bit history 
- https://github.com/magento/magento2/pull/29408#pullrequestreview-502971738
- https://github.com/magento/magento2/pull/29408#issuecomment-707683245

Reply:
It will set the website id and save the store model.
Alternative solution to avoid too general array type is list all data arguments separately, but that approach is not extensible.